From 3295a0ec01b247165b5998a6e9624732a552e88d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 19 Mar 2019 20:08:46 +0100
Subject: [PATCH 1/1] efi_selftest: fix test_hii_string_get_string()

The check testing the string result of get_string() returned the wrong
result. The result was ignored.

Use efi_st_strcmp_16_8() for the string comparison.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_selftest_hii.c | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

diff --git a/lib/efi_selftest/efi_selftest_hii.c b/lib/efi_selftest/efi_selftest_hii.c
index 8a0b3bc3536..f4b70f79508 100644
--- a/lib/efi_selftest/efi_selftest_hii.c
+++ b/lib/efi_selftest/efi_selftest_hii.c
@@ -783,19 +783,10 @@ static int test_hii_string_get_string(void)
 		goto out;
 	}
 
-#if 1
-	u16 *c1, *c2;
-
-	for (c1 = string, c2 = L"Japanese"; *c1 == *c2; c1++, c2++)
-		;
-	if (!*c1 && !*c2)
-		result = EFI_ST_SUCCESS;
-	else
-		result = EFI_ST_FAILURE;
-#else
-	/* TODO: %ls */
-	efi_st_printf("got string is %s (can be wrong)\n", string);
-#endif
+	if (efi_st_strcmp_16_8(string, "Japanese")) {
+		efi_st_error("get_string returned incorrect string\n");
+		goto out;
+	}
 
 	result = EFI_ST_SUCCESS;
 
-- 
2.20.1

